spring-cloud-aws-autoconfigure/src/main/java/io/awspring/cloud/autoconfigure/s3/S3Properties.java

```diff
@@ -0,0 +1,124 @@
+ * limitations under the License.
+ */
+
+package io.awspring.cloud.autoconfigure.s3;
+
+import io.awspring.cloud.autoconfigure.AwsClientProperties;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * Properties related to AWS S3.
+ *
+ * @author Maciej Walkowiak
+ */
+@ConfigurationProperties(prefix = S3Properties.PREFIX)
+public class S3Properties extends AwsClientProperties {
+
+	/**
+	 * The prefix used for S3 related properties.
+	 */
+	public static final String PREFIX = "spring.cloud.aws.s3";
+
+	/**
+	 * Option to enable using the accelerate enedpoint when accessing S3. Accelerate
+	 * endpoints allow faster transfer of objects by using Amazon CloudFront's globally
+	 * distributed edge locations.
+	 */
+	private Boolean accelerateModeEnabled;
+
+	/**
+	 * Option to disable doing a validation of the checksum of an object stored in S3.
+	 */
+	private Boolean checksumValidationEnabled;
+
+	/**
+	 * Option to enable using chunked encoding when signing the request payload for
+	 * {@link software.amazon.awssdk.services.s3.model.PutObjectRequest} and
+	 * {@link software.amazon.awssdk.services.s3.model.UploadPartRequest}.
+	 */
+	private Boolean chunkedEncodingEnabled;
+
+	/**
+	 * Option to enable using the dualstack endpoints when accessing S3. Dualstack should
+	 * be enabled if you want to use IPv6.
+	 */
+	private Boolean dualstackEnabled;
+
+	/**
+	 * Option to enable using path style access for accessing S3 objects instead of DNS
+	 * style access. DNS style access is preferred as it will result in better load
+	 * balancing when accessing S3.
+	 */
+	private Boolean pathStyleAccessEnabled;
+
+	/**
+	 * If an S3 resource ARN is passed in as the target of an S3 operation that has a
+	 * different region to the one the client was configured with, this flag must be set
+	 * to 'true' to permit the client to make a cross-region call to the region specified
+	 * in the ARN otherwise an exception will be thrown.
+	 */
+	private Boolean useArnRegionEnabled;
+
+	public Boolean getAccelerateModeEnabled() {
+		return accelerateModeEnabled;
+	}
+
+	public void setAccelerateModeEnabled(Boolean accelerateModeEnabled) {
+		this.accelerateModeEnabled = accelerateModeEnabled;
+	}
+
+	public Boolean getChecksumValidationEnabled() {
+		return checksumValidationEnabled;
+	}
+
+	public void setChecksumValidationEnabled(Boolean checksumValidationEnabled) {
+		this.checksumValidationEnabled = checksumValidationEnabled;
+	}
+
+	public Boolean getChunkedEncodingEnabled() {
+		return chunkedEncodingEnabled;
+	}
+
+	public void setChunkedEncodingEnabled(Boolean chunkedEncodingEnabled) {
+		this.chunkedEncodingEnabled = chunkedEncodingEnabled;
+	}
+
+	public Boolean getDualstackEnabled() {
+		return dualstackEnabled;
+	}
+
+	public void setDualstackEnabled(Boolean dualstackEnabled) {
+		this.dualstackEnabled = dualstackEnabled;
+	}
+
+	public Boolean getPathStyleAccessEnabled() {
+		return pathStyleAccessEnabled;
```

Comment:
```suggestion
		return this.pathStyleAccessEnabled;
```

Reply:
This is not a convention we used neither in 2.x nor 3.x. We can discuss (at this point i don't see value) but if so we would enforce it in checkstyle.